probe/websocket/ws_test.go

```diff
@@ -50,6 +51,21 @@ func TestWSPing(t *testing.T) {
 		}
 	}()
 
+	// wait for http server to start
+	conn, err := net.Conn(nil), error(nil)
+	for i := 0; i < 30; i++ {
```

Comment:
Consider adding a timeout for the waiting loop to avoid potential infinite loops if the server fails to start. [important]